stan/math/rev/mat/fun/divide.hpp

```diff
@@ -9,18 +9,162 @@
 namespace stan {
 namespace math {
 
+namespace internal {
+template <int R, int C>
+class matrix_scalar_divide_dv_vari : public vari {
+ public:
+  int rows_;
+  int cols_;
+  vari* adjCRef_;
+  vari** adjResultRef_;
+  double invc_;
+
+  explicit matrix_scalar_divide_dv_vari(const Eigen::Matrix<double, R, C>& m,
+                                        const var& c)
+      : vari(0),
+        rows_(m.rows()),
+        cols_(m.cols()),
+        adjCRef_(c.vi_),
+        adjResultRef_(ChainableStack::instance_->memalloc_.alloc_array<vari*>(
+            m.rows() * m.cols())),
+        invc_(1.0 / c.val()) {
+    Eigen::Matrix<double, R, C> result = invc_ * m;
+    Eigen::Map<matrix_vi>(adjResultRef_, rows_, cols_)
+        = result.unaryExpr([](double x) { return new vari(x, false); });
```

Comment:
Super minor thing that I noticed, but you could also write this as:
```
Eigen::Map<matrix_vi>(adjResultRef_, rows_, cols_)
= (invc_ * m).unaryExpr([](double x) { return new vari(x, false); })
```
To avoid the creation of the ```result``` temporary

(Not a review, feel free to ignore)

Reply:
@andrjohns:  Thanks --- that sounds like a good idea.

@peterwicksstringfield :  This is optional---we have a general policy of not double reviewing or unapproving unless absolutely necessary for a bug.  Let me know if you'd like to make this update or just merge.

Reply:
Thanks, that transformation avoids collapsing an expression template.